core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -300,11 +304,11 @@ public void disableStore(String storeType) {
                result.add((T) real);
             }
          }
-         for (CacheWriter w : writers) {
+         getWritersStream().forEach(w -> {
```

Comment:
You could avoid creating some extra stream instances with a `forEachWriter` method that calls `writers.forEach()` and then `txWriters.forEach()`